### ISSUE TITLE
[feature request] Plotting additional info collected by monitor using results_plotter

### ISSUE THREAD

**Issue by nisheeth-golakiya** [hill-a/stable-baselines#872]
The current [`results_plotter`](https://github.com/hill-a/stable-baselines/blob/master/stable_baselines/results_plotter.py#L48) only plots episode rewards. Although [`TensorBoardCallback`](https://stable-baselines.readthedocs.io/en/master/guide/tensorboard.html#logging-more-values) can track additional variables, it would be nice to be able to select which metric(e.g. `info1` in the code below) to plot from the csv file created by the monitor.

Additionally, the csv file created by the monitor (in case of vectorized environments) has a messy index column as shown below:
```python
import random

import gym
import numpy as np
from stable_baselines.common.vec_env import SubprocVecEnv
from stable_baselines.common.cmd_util import make_vec_env
from stable_baselines.bench import load_results


class CustomEnv(gym.Env):
    def __init__(self):
        super(CustomEnv, self).__init__()
        self.observation_space = gym.spaces.Box(low=0.0, high=1.0, shape=(3,))
        self.action_space = gym.spaces.Discrete(3)

    def reset(self):
        obs = self._get_obs()
        return obs

    def step(self, action):
        obs = self._get_obs()
        reward = np.random.randn()
        done = random.choice([True, False])
        info = {'info0': np.random.randn(), 'info1': np.random.randn()}
        return obs, reward, done, info

    def _get_obs(self):
        return np.random.random((3,))


if __name__ == "__main__":
    n_envs = 4
    env = make_vec_env(CustomEnv,
                       n_envs,
                       monitor_dir='./log/',
                       vec_env_cls=SubprocVecEnv,
                       vec_env_kwargs={"start_method": 'fork'})
    obs = env.reset()

    for i in range(100):
        action = [env.action_space.sample() for _ in range(n_envs)]
        obs, reward, done, info = env.step(action)

    env.close()

    df = load_results('./log')
    print(df.head())


```
Output:
```bash
   index         r  l         t
0      0  0.266806  1  0.014221
1      1 -0.191954  1  0.014738
2      0  0.074852  2  0.014752
3      1 -1.529633  1  0.015202
4      0  2.770444  4  0.015610

```
It looks like these index values are from the individual csv files in the log folder, but there seems to be no use of this when they are combined into one dataframe.

Other minor concern is to pass the `info_keywords` in [`make_vec_env`](https://github.com/hill-a/stable-baselines/blob/master/stable_baselines/common/cmd_util.py#L60) method (as `monitor_kwargs` maybe?). It would make things easier so that we don't have to rewrite `make_env` function by ourselves.

**Comment by Miffyli** [comment]
`results_plotter`: I am not sure how in-depth we want to modify these, they seem like older pieces of code. I think utility functions for reading monitor files would be a better fit (e.g. `load_monitor(path_to_monitor) -> dict(column_name: list of values)`).

Passing `info_keywords`: Good suggestion! It sounds simple enough we could add it to both here and stable-baselines3. @araffin comments? This should be just passing few parameters env creation functions.

As for the monitor files with vec-envs: I am not quite sure what is the issue. Indeed they get mixed up like that, and you can not separate which env created which lines. However if you wish to track e.g. average episodic reward, this should not be a problem if all the environments are the same (just independent instances of same env). An alternative I have used is to create a new monitor file per environment with unique id, which then can be interleaved later.

**Comment by nisheeth-golakiya** [comment]
> As for the monitor files with vec-envs: I am not quite sure what is the issue.

If we want to work with the dataframe returned by `load_results`, it might create confusion. (`df.index` will mean the pandas RangeIndex object but `df['index']` will mean the messy column named 'index'). I believe my suggestion is to drop this column named 'index' when it is returned by `load_results`.

**Comment by araffin** [comment]
> I think utility functions for reading monitor files would be a better fit 

yes, we already have that https://stable-baselines3.readthedocs.io/en/master/common/monitor.html#stable_baselines3.common.monitor.load_results

>has a messy index column as shown below:

What did you expect? As environment are run in parallel, there is one monitor file per environment (see `make_vec_env` code) that is merged into one when calling `load_results`. The time is used to sort them.

>I believe my suggestion is to drop this column named 'index' when it is returned by `load_results`

Please do a PR for that ;)

> It would make things easier so that we don't have to rewrite make_env function by ourselves.

Actually, you can use the zoo to define a monitor wrapper that will record what you want. But yes, it only few lines to redefine it.

**Comment by nisheeth-golakiya** [comment]
>The current results_plotter only plots episode rewards. Although TensorBoardCallback can track additional variables, it would be nice to be able to select which metric(e.g. info1 in the code below) to plot from the csv file created by the monitor.

Regarding this, my point was to plot variables other than episodic reward. [Here](https://github.com/hill-a/stable-baselines/blob/master/stable_baselines/results_plotter.py#L59) in the code, instead of `y_var = timesteps.r.values`, if we could have something like `y_var = timesteps[ylabel].values`, where `ylabel` can be passed into the function, it would make the code more generalized.
Sorry I couldn't explain it clearly earlier.
As @Miffyli has suggested, if the code is too old to modify, that is also understandable. 

>Please do a PR for that ;)

Sure!

>Actually, you can use the zoo to define a monitor wrapper that will record what you want. But yes, it only few lines to redefine it.

Okay, I understand.
